src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js

```diff
@@ -660,7 +679,7 @@ ApiGlobe.prototype.setRange = function setRange(pRange, isAnimated) {
 
     return this.scene.currentControls().setRange(pRange, isAnimated).then(() => {
         this.scene.notifyChange(1);
-        this.setSceneLoaded().then(() => {
+        return this.setSceneLoaded().then(() => {
             this.scene.currentControls().updateCameraTransformation();
             this.viewerDiv.dispatchEvent(eventRange);
```

Comment:
try with `then` promise,  you do getRange() just after setRange(), While the setRange is not yet completed 

setRange(10000).then(() => getRange())